### PR TITLE
Migrate the contents of TLS.md into README.md

### LINKED ISSUE #888: Migrate TLS.md into README.md
While reviewing some old files, we noticed https://github.com/valkey-io/valkey/blob/unstable/TLS.md should probably be in the main readme instead of a separate file. There are also two todo items, we should get if those are implemented (I think at least benchmark is) but need to double check out the other one. If they are, we can close them, otherwise let's make issues.

### PR DESCRIPTION
**Pull request by vax-r**
Migrate the contents in TLS.md into TLS sections including building,
running and detail supports. TODO list in the TLS.md is almost done
except the implementation of benchmark support is still not the best
approach which should migrate to hiredis async mode.

Closes #888

### CODE REVIEW

**Review by zuiderkwast — COMMENTED** [review]
Great! Did you move all content of TLS.md? If yes, then also delete TLS.md. The todo list belongs in github issues, not in the md files.

File: README.md [hunk 1]
```diff
@@ -158,8 +166,55 @@ line, with exactly the same name.
...
+Use `valkey-cli` to connect to the Valkey server:
+```
+./src/valkey-cli --tls \
+    --cert ./tests/tls/valkey.crt \
+    --key ./tests/tls/valkey.key \
+    --cacert ./tests/tls/ca.crt
+```
+
+Specifying `--tls-replication yes` can make a Replica connect to the master.
+Using `--tls-cluster yes` to make Valkey Cluster use TLS across nodes.
```
Comment (zuiderkwast) [inline README.md]:
```suggestion
Specifying `--tls-replication yes` makes a replica connect to the primary.

Using `--tls-cluster yes` makes Valkey Cluster use TLS across nodes.
```

File: README.md [hunk 2]
```diff
@@ -158,8 +166,55 @@ line, with exactly the same name.
...
+
+Currently, synchronize I/O for TLS is implemented in a hackish way, involving blocking sockets and socket-level timeouts, leading to potential inaccuracies and syscall overhead. Transitioning entirely to pure async I/O may be a better solution, especially for replication. Although cluster keys migration could be more challenging, it presents an opportunity for improvement.
+
+#### TLS Multi-port support
+Consider the implications of allowing TLS to be configured on a separate port, making Valkey listening on multiple ports:
+
+1. Startup banner port notification
+2. Proctitle
+3. How slaves announce themselves
+4. Cluster bus port calculation
```
Comment (zuiderkwast) [inline README.md]: Threaded I/O now supports TLS, so this is outdated. The other internals don't need to be in the README. I believe it's mostly outdated or irrelevant and the relevant information can be found in the source code, including code comments.

TLS multi-port section is outdated. All of this is solved.

So let's delete this.

```suggestion
```

**Review by zuiderkwast — APPROVED** [review]
Good, let's merge this. I'll take care of deleting the TLS.md file, so we don't have duplicated information around.

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**